src/pages/iou/request/step/IOURequestStepDistance.tsx

```diff
@@ -87,6 +89,7 @@ function IOURequestStepDistance({
             },
         [optimisticWaypoints, transaction],
     );
+    const {shouldFetchRoute, validatedWaypoints} = useFetchRoute(transaction, waypoints, action);
```

Comment:
Strong reason to believe that this change might've let to the surfacing of the following issue:
- https://github.com/Expensify/App/issues/48630

which in order to fix we had to extend the logic of `useFetchRoute` and involve BE. More details in the issue and [proposed solution](https://github.com/Expensify/App/issues/48630#issuecomment-2365034676).